src/EnergyPlus/DataMoistureBalance.hh

```diff
@@ -53,7 +53,6 @@
 #include <ObjexxFCL/Array2D.hh>
 #include <ObjexxFCL/Array3D.hh>
 #include <ObjexxFCL/Array4D.hh>
-#include <ObjexxFCL/Array5D.hh>
```

Comment:
Buh-bye.

Reply:
Thanks. It's hard to have good code with bad data structures (unfortunately, the inverse is not true). One of the tragedies of Fortran is that multi-dimensional matrices are stored in column-major order, meaning the outer indices are to the right (as if they are in Hebrew). There are no multi-dimensional arrays in C/C++, but because English and therefore C/C++ are left-to-right any nested access idiom will have the outer indices on the left.

When EnergyPlus was translated to Fortran, this flip caused performance problems. The hotspots were dealt with shortly after the translation process (you will see the comment `// tuned/transposed` scattered throughout the code, that's what it refers to). But many smaller hotspots and bad loops remain. Cleaning that up is leading to better data structures and better code overall.